universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -0,0 +1,184 @@
+package services.program;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import models.Program;
+import play.db.ebean.Transactional;
+import repository.ProgramRepository;
+import services.question.QuestionDefinition;
+
+public class ProgramServiceImpl implements ProgramService {
+
+  private ProgramRepository programRepository;
+
+  @Inject
+  public ProgramServiceImpl(ProgramRepository programRepository) {
+    this.programRepository = checkNotNull(programRepository);
+  }
+
+  @Override
+  public ImmutableList<ProgramDefinition> listProgramDefinitions() {
+    return listProgramDefinitionsAsync().toCompletableFuture().join();
+  }
+
+  @Override
+  public CompletionStage<ImmutableList<ProgramDefinition>> listProgramDefinitionsAsync() {
+    return programRepository
+        .listPrograms()
+        .thenApply(
+            programs ->
+                programs.stream()
+                    .map(program -> program.getProgramDefinition())
+                    .collect(ImmutableList.toImmutableList()));
+  }
+
+  @Override
+  public Optional<ProgramDefinition> getProgramDefinition(long id) {
+    return getProgramDefinitionAsync(id).toCompletableFuture().join();
+  }
+
+  @Override
+  public CompletionStage<Optional<ProgramDefinition>> getProgramDefinitionAsync(long id) {
+    return programRepository
+        .lookupProgram(id)
+        .thenApply(optionalProgram -> optionalProgram.map(Program::getProgramDefinition));
+  }
+
+  @Override
+  public ProgramDefinition createProgramDefinition(String name, String description) {
+    Program program = new Program(name, description);
+    return programRepository.insertProgramSync(program).getProgramDefinition();
+  }
+
+  @Override
+  @Transactional
```

Comment:
suggestion: Looking at [these docs](https://ebean.io/docs/transactions/transactional), it seems like to get the benefit of `@Transactional`, we may need to add an `ebean.mf` manifest file with the `transaction-packages` set.